### PR TITLE
Fix delete all goals

### PR DESCRIPTION
**Pull request by devbeoservice**
-Added a delete all button with confirmation (github style) where user has to write 'DELETE' to get it done.
-Added axios to be required in web
-made a change in gitignore for convenience while developing, ignoring docker-compose.yaml

### CODE REVIEW

**Comment by devbeoservice** [comment]
fixed as per requirements from https://discord.com/channels/503244944807559178/1237085818103922799/1290684943785005108 and https://discord.com/channels/503244944807559178/1237085818103922799/1290689245995995219

File: web/src/components/game/goals/GoalManagement.tsx [hunk 1]
```diff
@@ -149,6 +196,38 @@ export default function GoalManagement() {
                     />
                 </DialogContent>
             </Dialog>
+
+            {/* Delete Confirmation Dialog */}
+            <Dialog ref={deleteDialogRef}>
```
Comment (cjs8487) [inline web/src/components/game/goals/GoalManagement.tsx:201]: If we're going to seat multiple dialogs into a single component we should either
a) make them each their own component, and utilize `useImperativeHandle`/`forwardRef` to isolate state/functionality
b) Use a modular composing system for the content so we only seat a single dialog and have a single ref, such as something like the following
```tsx
const [dialogContent, setDialogContent] = useState(null)
const launchDelateAllConfirmation = () => {
    setDialogContent(<DeleteAllConfirmationDialog />)
    dialogRef.current.open()
}
...
return (
    ...
    <Button onClick={launchDeleteAllConfirmation}
    <Dialog ref={dialogRef}>
        {dialogContent}
    </Dialog>
)
```

Comment (cjs8487) [inline web/src/components/game/goals/GoalManagement.tsx:201]: If a dialog requires it's own standalone state, it should usually be it's own component

File: web/src/components/game/goals/GoalManagement.tsx [hunk 2]
```diff
@@ -35,8 +37,38 @@ export default function GoalManagement() {
...
+                method: 'DELETE',
+            });
+    
+            if (!response.ok) {
+                // Handle failure silently
+                return;
+            }
+    
+            // Refresh the goal list if needed
+            mutateGoals();
```
Comment (cjs8487) [inline web/src/components/game/goals/GoalManagement.tsx:64]: We should probably call `mutateGoals` in the finally block in order to catch corner cases where we get partial deletion before running into a database error or similar

File: api/src/routes/games/Games.ts [hunk 3]
```diff
@@ -285,4 +285,28 @@ games.get('/:slug/permissions', async (req, res) => {
     });
 });
 
+games.delete('/:slug/delete-all-goals', async (req, res) => {
```
Comment (cjs8487) [inline api/src/routes/games/Games.ts:288]: Route names should be camelCased if they are multiple words

File: api/src/database/games/Goals.ts [hunk 4]
```diff
@@ -93,3 +93,19 @@ export const gameForGoal = async (goalId: string) => {
 
     return goal?.game;
 }
+
+export const deleteAllGameGoals = async (gameSlug: string) => {
```
Comment (cjs8487) [inline api/src/database/games/Goals.ts:97]: Name this deleteAllGoalsForGame

File: web/src/components/game/goals/GoalManagement.tsx [hunk 5]
```diff
@@ -35,8 +37,38 @@ export default function GoalManagement() {
...
+        }
+    
+        try {
+            setLoading(true);
+            const response = await fetch(`/api/games/${slug}/delete-all-goals`, {
+                method: 'DELETE',
+            });
+    
+            if (!response.ok) {
+                // Handle failure silently
```
Comment (cjs8487) [inline web/src/components/game/goals/GoalManagement.tsx:59]: This failure doesn't necessarily need to be silent, you can toast the message sent by the server if you feel it is appropriate (and safe) to do so. Generally, failure cases where the failure is not obvious should not fail silently

File: web/src/components/game/goals/GoalManagement.tsx [hunk 6]
```diff
@@ -128,8 +160,23 @@ export default function GoalManagement() {
...
+                {canModerate && (
+                    <Box sx={{ display: 'flex', justifyContent: 'flex-start', width: '100%', mt: 3 }}>
+                        <Button
+                            onClick={() => deleteDialogRef.current?.open()} // Opens delete confirmation dialog
+                            disabled={loading}
+                            color="error"
+                            variant="contained"
+                            sx={{ maxWidth: '200px' }}
+                        >
+                            {loading ? 'Deleting...' : 'Delete All Goals'}
```
Comment (cjs8487) [inline web/src/components/game/goals/GoalManagement.tsx:172]: Having the delete dialog stay open, and having this loading/processing indicator feels redundant

**Review by cjs8487 — CHANGES_REQUESTED** [review]
Overall comments
- SnackbarNotifier should be deleted
- The delete all button should probably be colocated with other controls at the top of the activity, along with the settings and upload buttons, especially since upload and delete all will often be used in tandem (at least until a "replace all" option is added for upload)
- Delete all should probably not be a contained button, though the error styling is semantically correct

**Comment by devbeoservice** [comment]
resolved all except location of the button

**Review by cjs8487 — APPROVED** [review]

**Pull request merged**